assets/iframe.html.tmpl

```diff
@@ -0,0 +1,88 @@
+<!-- @ts-nocheck -->
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8">
+  <title>Mattermost DevSecOps</title>
+  <meta name="viewport" content="width=device-width, height=device-height, initial-scale=1.0">
+  <script
+    src="https://res.cdn.office.net/teams-js/2.34.0/js/MicrosoftTeams.min.js"
+    integrity="sha384-brW9AazbKR2dYw2DucGgWCCcmrm2oBFV4HQidyuyZRI/TnAkmOOnTARSTdps3Hwt"
+    crossorigin="anonymous"
+  ></script>
+</head>
+<body>
+    <iframe
+        style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; border: none;"
+        title="Mattermost DevSecOps"
+        src="about:blank">
+    </iframe>
+  <script>
+    var iframe = document.querySelector('iframe');
+
+    // Initialize the Microsoft Teams SDK
+    microsoftTeams.app.initialize(['{{.SiteURL}}']).then(() => {
+      microsoftTeams.app.notifySuccess();
+    }).catch((error) => {
+      console.error('Failed to initialize Microsoft Teams SDK:', error);
+    });
+
+    const expectedTenantId = '{{.TenantID}}';
+    const domainRoot = '{{.SiteURL}}';
+
+    // Choose the iFrame content based on the tenant.
+    microsoftTeams.app.getContext().then((context) => {
+      let tenantId;
+      if (context && context.user && context.user.tenant && context.user.tenant.id) {
+        tenantId = context.user.tenant.id;
+      }
+
+      // If the expected tenant matches the actual tenant then try to get auth token
+      if (tenantId === expectedTenantId) {
+        // Build query params to be sent to the iframe.
+        const params = new URLSearchParams()
+        params.set('app_id', context.app.appId.appIdAsString);
+  
+        // Extract the subPageId (subEntityId coming from the Microsoft Teams SDK User Activity notification)
+        // and send it to the iframe to redirect the user to what triggered the notification.
+        if (context && context.page && context.page.subPageId) {
+          params.set('sub_entity_id', context.page.subPageId);
+
+          // Since we received a sub_entity_id, redirect to the subEntityId page manually using the Microsoft Teams SDK.
+          // This is a workaround so we send the user to the tab application when opening the link from the 
+          // Microsoft Teams User Activity notifications page.
+          // Trying to navigate to the app within the app seems to be a no-op from the SDK, which is good for us.
+          // microsoftTeams.pages.navigateToApp({
+          //   appId: context.app.appId.appIdAsString,
+          //   subPageId: context.page.subPageId
+          // }).then(() => {
+          //   console.log("Successfully opened link, redirecting to the tab application.");
+          // }).catch((error) => {
+          //   console.error('Failed to open link, fallback to using the iframe.', error);
+          // });
```

Comment:
We can probably remove all this, since we have an intermediate notification page.

```suggestion
```